omod/src/main/webapp/resources/js/aijar.js

```diff
@@ -349,3 +351,56 @@ var fieldHelper = {
     }
 };
 
+function blockEncounterOnSameDateEncounter(encounterDate, instruction) {
+
+    if (!(instruction == 'block' || instruction == 'warn'))
+        return;
+
+    var date = jq(encounterDate).val();
+    var formId = jq('[name=htmlFormId]').val();
+    var patientId = jq('[name=personId]').val();
+
+    if (jq('[name=encounterId]').val() == null) {
+        jq.get(
+            getContextPath() + '/module/htmlformentry/lastEnteredForm.form',
+            {formId: formId, patientId: patientId, date: date, dateFormat: 'yyyy-MM-dd'},
+            function (responseText) {
+
+                if (responseText == "true") {
+                    if (instruction == "warn") {
+
+                        // get the localized warning message and display it
+                        jq.get(getContextPath() + "/module/htmlformentry/localizedMessage.form",
+                            {messageCode: "htmlformentry.error.warnMultipleEncounterOnDate"},
+                            function (responseText) {
+                                jq().toastmessage('showWarningToast', responseText);
+                            }
+                        );
+
+                    } else if (instruction == "block") {
+
+                        // get the localized blocking message and display it
+                        jq.get(getContextPath() + "/module/htmlformentry/localizedMessage.form",
+                            {messageCode: "htmlformentry.error.blockMultipleEncounterOnDate"},
+                            function (responseText) {
+                                jq().toastmessage('showWarningToast', responseText);
+                                var parameters = window.location.search.split("&", 2);
+                                var url = window.location.origin + "/" + OPENMRS_CONTEXT_PATH + "/" + "coreapps/patientdashboard/patientDashboard.page" + parameters[0] + "&" + parameters[1] + "&";
+                                setTimeout(function () {
+                                    window.location.href = url;
+                                }, 2000);
+
+                            }
+                        );
+
+                        //clear the date and continue entering the form
+                        jq(encounterDate).val('');
+                    }
+                } else {
```

Comment:
@slubwama this else is empty so is not needed

Reply:
@ssmusoke  was this PR Complete seems like there are comments that were not attended to??